### DISCUSSION TITLE
Is it possible to modify the data before filling in the field whith relationship in a fieldset?

### DISCUSSION THREAD

**Discussion by edson36** [filamentphp/filament#13079] [Help] [answered]
### Package

Form builder

### Package Version

v3

### How can we help you?

I'm having a problem, I have a hasOne relationship in a fieldset, I need to format some data within the relationship before showing it in the field and I can't do it at all, not by afterStateHydrated in the field, ->mutateRelationshipDataBeforeFillUsing in the relationship, nor by mutateFormDataBeforeFill in the edit.
When debugging the modified data, it is correct, but when filling in it appears unchanged.

Is it possible to modify the data before filling in the field?

**Comment by agustavo87** [comment, marked as answer]
Hi, I had the same issue and, luckily, solved it. The problem was that I hade _multiple_ `Fieldset` with the _same_ relationship. It seems that Filament hydrate all the properties in `data.relatedModel` for the same relationship each time the `->relationship('relatedModel')` is called. So any change you made is overridden each time it is rehydrated from the original model. So the solution was simply in my case, to use `Group::make()->relationship('relatedModel')` once and then inside, in the `schema([...])` the `Fieldset` for the related model, withouth calling `relationship()` each time, and all the standard hooks worked as expected.

For example. If I had something like:

```php
Fieldset::make('Related Model)
   ->relationship('relatedModel')
   ->schema([...]),
Fieldset::make('Other fields of related model)
   ->relationship('relatedModel')
   ->schema([...])
```

Then the refactor was simply

```php
Group::make()
  ->relationship('relatedModel')
  ->schema([
    Fieldset::make('Related Model)
       ->schema([...]),
    Fieldset::make('Other fields of related model)
       ->schema([...])
  ])
```
